introduce ability to favorite a tool

martenson opened the pull request:
* Create tool favoriting API that can be easily expanded to other objects.
* Create button on every toolform to favorite/unfavorite tool.

~~I wanted to use `axios.delete` but had trouble with Routes parsing the wildcard patterns when given the tool_id encoded in the url. Would still be better API I think if anyone has an idea how to do it. ping @mvdbeek~~

martenson committed:
introduce ability to favorite a tool

martenson committed:
introduce ability to favorite a tool

anilthanki committed:
adds Aequatus visualisation plugin

updates Aequatus control panel-settings font size

Update README.md

fix path to aequatus static scripts

updates with upstream

removes console

removes installation guide

removes demo

removes duplicates

removes d3.js

removes bootstrap

prunes font-awesome

removes less of font-awesome

removes scss of font-awesome

uses font-swesome min

removes bootstrap

removes leftandright css class

updates jQuery and changes css class for popup

updates jQuery

updates fontawesome to 5.6.1

restore galaxy.yml.sample

add missing resource

update paths after updating font

remove console log

switch to minified jquery ui

mvdbeek commented on lib/galaxy/webapps/galaxy/buildapp.py:
```diff
@@ -628,6 +628,18 @@ def populate_api_routes(webapp, app):
+                          '/api/users/{id}/favorites/tools',
+                          controller='users',
+                          action='remove_favorite_tool',
+                          conditions=dict(method=["POST"]))
```
DELETE should work after https://github.com/galaxyproject/galaxy/pull/6467 is merged, since it upgrades to recent WebOb, which introduced DELETE with a body in version 1.7 https://docs.pylonsproject.org/projects/webob/en/stable/changes.html. But wouldn't it be better to have the route be `/api/users/{id}/favorites/tools/{tool_id}` ? Than POST and DELETE would feel pretty natural to me.

martenson replied:
>trouble with Routes parsing the wildcard patterns when given the tool_id encoded in the url

martenson replied:
tried few more things and it seems `/{object_id:.*?}` works, so I will refactor this as well

martenson replied:
This is now in.

guerler commented on lib/galaxy/webapps/galaxy/api/users.py:
```diff
@@ -510,6 +510,38 @@ def set_information(self, trans, id, payload={}, **kwd):
         return {'message': 'User information has been saved.'}
 
+    @expose_api
+    def set_favorite_tool(self, trans, id, payload={}):
```
I think we should generalize this slightly such that we wont have to add individual endpoints for other `favorite` items e.g. workflows, visualizations etc. by creating an endpoint `set_favorites` instead.

martenson replied:
Sounds good to me, I can do that.

martenson replied:
This is now in.

martenson committed:
refactor and enhance the API to be generic favorite endpoint

martenson committed:
refactor and enhance the API to be generic favorite endpoint

martenson committed:
refactor the favorite API to use DELETE method for removal

martenson committed:
refactor the favorite API to use DELETE method for removal

martenson committed:
add basic API tests for favoriting

and fix errors caught by the tests

martenson committed:
add basic API tests for favoriting

and fix errors caught by the tests

natefoo committed:
WIP job assignment using database transaction isolation

natefoo committed:
Rename job assigner to job grabber and clean up, also add support for
(preferable) FOR UPDATE ... SKIP LOCKED

jmchilton committed:
Add to_dict() to ObjectStores.

Dictify an object store configuration back to its to config_dict description.

guerler committed:
Add reload feature to display applications api endpoint

guerler committed:
Add display application vue component

guerler committed:
Add display application vue component

guerler committed:
Fix spacing in admin app router

guerler committed:
Load display application data and display number of available applications

guerler committed:
Remove legacy display applications mako and controller endpoints

guerler committed:
Submit reload requests to new display application api endpoint

guerler committed:
Remove debug outputs

guerler committed:
Use bootstrap vues table component to render display applications table

guerler committed:
Highlight reloaded/failed display application rows individually

guerler committed:
Fix linting

guerler committed:
Fix selenium test case for display application visibility

natefoo committed:
Integrate DB serialization methods with new handler assignment process

mvdbeek committed:
Allow passing unused keywords to container resolver

mvdbeek committed:
Merge container resolver with high-level dependency manager

mvdbeek committed:
Introduce a ContainerDependency class

mvdbeek committed:
Adjust _requirements_to_dependencies_dict to also report available containers in the manage dependencies menu

mvdbeek committed:
Augment NullDependency with version

and always record incoming null dependencies,
since the container resolvers don't return
NullDependency objects.

mvdbeek committed:
Fix matching single package mulled docker containers

An image like `quay.io/biocontainers/ucsc-liftover:357--1` would
otherwise have the package name `quay.io/biocontainers/ucsc-liftover`
instead of `ucsc-liftover`

guerler committed:
Fix selenium test, add missing applications message

guerler committed:
Show warning message after requesting display applications

mvdbeek committed:
Only walk through enabled container resolvers

guerler committed:
Separate api requests from vue component for admin services

guerler committed:
Fix comment header for display applications api

jdavcs committed:
Enable tagging+annotation of jobs regardless of state

See issue #6330

Nerdinacan committed:
Added initialization queue, reformatted tempaltes for single entry point, global removals

Nerdinacan committed:
Python linting fixes

mvdbeek committed:
Avoid circular imports

by moving {App,Job,Tool} Info classes to dependencies.py

mvdbeek committed:
Add `can_uninstall_dependencies` attribute to ContainerResolver

Nerdinacan committed:
Merge branch 'dev' into init_queue_2

mvdbeek committed:
Make DependencyManager container & destination-aware

This will allow us to use the proper parameters when pulling containers.

guerler committed:
Add require admin to display applications api

mvdbeek committed:
Allow specifying resolver index and resolver_type when installing dependencies via tools API

mvdbeek committed:
Drop NullDependency only when not skipping resolver

mvdbeek committed:
Mention index and resolver_type in API docstring

mvdbeek committed:
Move Container classes into container_classes to avoid circular imports

mvdbeek committed:
Make mulled docker containers installable

guerler committed:
Move error stack controller endpoint to tools api

mvdbeek committed:
Allow selecting resolver when uninstalling through tools API

mvdbeek committed:
Add integration test for un/install with resolver_type

Nerdinacan committed:
Removed global functionality out of masthead

Nerdinacan committed:
Removed separate stylesheet

guerler committed:
Add error stack vue component and client side routes

guerler committed:
Remove legacy tool error mako file

guerler committed:
Remove unused attribute

guerler committed:
Remove unused tag in vue component

guerler committed:
Allow sorting by file name in error log table

guerler committed:
Fix comment section of error stack api endpoint

natefoo committed:
Add support for dynamic handlers when using DB serialization handler assignment

natefoo committed:
SQLAlchemize the grabber query and fold grabber directly into JobHandlerQueue

mvdbeek committed:
Use singularity build with biocontainers

This makes it possible to pull a container from
quay.io/biocontainers and to convert it to singularity format
for later execution (via the CachedMulledSingularityContainerResolver).

This is nicely integrated with the DependencyManager view, so
you can (as for conda or docker) do:
```
requests.post(galaxy_url+'/api/tools/deseq2/dependencies', {'key': API_KEY, 'resolver_type': 'mulled_singularity'})
```

This is quite lightweight, the resulting image is significantly smaller
than the conda env (since it is compressed), and the install process
is much faster.

mvdbeek committed:
Try cached resolver before non-cached

guerler committed:
Remove admin requirment from index endpoint of display applications

mvdbeek committed:
Add integration test for singularity resolution

jmchilton committed:
Remove redundant Task code.

get_job is overridden in Task, that should be sufficient I suspect.

jmchilton committed:
Reuse get_param_dict in job wrapper.

There is a comment to this, listen to the smart comment and then delete.

mvdbeek committed:
Skip singualrity integration tests if singularity not available

jmchilton committed:
Test for 'metadata' in newer style discovered primary datasets.

natefoo committed:
No longer using trace logging

natefoo committed:
Fix for use of `log.trace` in Slurm runner

jmchilton committed:
Fix for testing extra files in tool tests.

Wouldn't be testing if primary file didn't specify a filename to test against.

jmchilton committed:
Tests for tool provided metadata specifing extra files.

martenson committed:
Merge pull request #7131 from jmchilton/extra_files_3

Fixes and tests for collecting extra files for discovered datasets.

dannon committed:
Minor formatting; tabstop=4

dannon committed:
Avoid using underscore where there's a vanilla alternative.

mvdbeek committed:
Add MulledSingularityContainerResolver to __all__

Otherwise speicifying the resolver in containers_conf.xml fails.

dannon committed:
Merge pull request #7115 from ic4f/feature_i6330

[WIP] Enable tagging+annotation of jobs regardless of state

guerler committed:
Use b-alert instead of basic bootstrap alert messages

guerler committed:
Replace default links with bootstrap vue buttons

jmchilton committed:
Remove unneeded session flushes and adds when collecting primary datasets.

Nerdinacan committed:
7132 Updated singleton to address cors issue in toolshed frames

guerler committed:
Use bootstrap vue message alerts

Nerdinacan committed:
Replaced test code with real code

mvdbeek committed:
Pass in user, fixes dependency install for data manager tools

mvdbeek committed:
Update testing docker image; drop mysql tests, add singularity

mvdbeek committed:
Drop tools without installable dependencies

These seem to be beyond legacy, and since their not installable let's
drop them from the default config. I left them in tool_conf.xml.main,
since they appear to be installed on main.

mvdbeek committed:
Do run through build_mulled resolver if install is set

mvdbeek committed:
Make NullDependencyManager a class with empty non-container dependencies

jmchilton committed:
Merge pull request #7136 from mvdbeek/remove_tools_without_deps

Drop tools without installable dependencies

mvdbeek committed:
Fix __str__ for BuildSingularityContainerResolver

mvdbeek committed:
Download a binary that works on OSX

jmchilton committed:
Remove collect_outputs_from option.

And just disable dynamic discovery of outputs from new_file_path for legacy profile tools.

This is part of an ongoing effort to kill new_file_path and simplify output discovery.

This option was added by @natefoo 10 years ago and he switched main to disable new_file_path discovery and noone ever complained. So lets just do this for all Galaxies - if main hasn't run tools that require this functionality in a decade - I very much doubt any Galaxies require this.

mvdbeek committed:
DependencyManager lint fix

mvdbeek committed:
Make `resolver` methods respect enabled_container_types

mvdbeek committed:
Don't fail if build exists and also don't remove build dir if it already exists

mvdbeek committed:
Add ExplicitSingularityContainerResolver

mvdbeek committed:
Make containers list, map can only be iterated once

mvdbeek committed:
Update Dockefile and run_test_wrapper.sh

We use the Galaxy users' home dir for HOME.
Otherwise os.environ.get('HOME') would return '/home/galaxy',
but jobs run would see /galaxy.

mvdbeek committed:
Fix container integration tests

mvdbeek committed:
Temporarily test from mvdbeek/testing-base

mvdbeek committed:
Only run integration tests with --privileged

guerler committed:
Use regular js-event handling instead of jquery for uploader

guerler committed:
Fix dragover highlighting for composite dataset uploader

guerler committed:
Use default js to handle drop events in tool form

dannon committed:
Client formatting.

dannon committed:
Merge branch '7132_fix_cors_singleton_storage' into dev

martenson committed:
Merge pull request #7087 from dfornika/mash-sketch-datatype

Add datatype for Mash Sketch files.

dannon committed:
Fixes toolshed and reports styling that was previously applied at the mako template level.  We'll want to update the look here, I think, but this is an incremental step that correctly uses the background image, bundled via webpack.

jmchilton committed:
Merge remote-tracking branch 'jmchilton/dev' into pulsar_resubmit

jmchilton committed:
Fixes for Pulsar error handling.

More uniform error hanlding to fix test. Simplify test case of resubmission to test just jobs resubmit properly (no need to test Pulsar-to-Pulsar resubmission in particular here I don't think, maybe as a second test - but not a first?).

jdavcs committed:
Correct code error in 'build job runner' tutorial

When  __init__ is called, **kwargs (most likely) will not contain
the 'runner_param_specs' key. And if so, calling update() on
kwargs['runner_param_specs'] will cause an error. Check if key exists,
then update or create.

Nerdinacan committed:
Merged dev... again

jdavcs committed:
Code style improvement to previous commit

dannon committed:
Merge pull request #7148 from ic4f/dev

Correct code error in 'build job runner' tutorial

Nerdinacan committed:
added store library to exported libs in bundleEntries

dannon committed:
Merge pull request #7097 from guerler/display_applications

Replace display applications mako with vue component

dannon committed:
Client format I intended to lump into display applications merge.

dannon committed:
Merge pull request #7134 from jmchilton/remove_flushes

Remove unneeded session flushes and adds when discovering outputs.

dannon committed:
Merge pull request #7149 from Nerdinacan/include_store_in_bundleEntries

added store library to exported libs in bundleEntries

guerler committed:
Merge branch 'dev' into admin_error_log

guerler committed:
Fix consistency of parameter naming to in error stack component

dannon committed:
Merge pull request #6484 from anilthanki/aequatus-vis

Adds Aequatus visualisation plugin

jmchilton committed:
lint fix

jmchilton committed:
Merge pull request #7053 from nuwang/pulsar_resubmit

Added resubmit handlers on Pulsar connection failures

guerler committed:
Fix connected value handling in workflow editor

guerler committed:
Remove pointer from inactive collapsible title

guerler committed:
Fix tooltip update for collapsible and connected input parameters in workflow editor

guerler committed:
Fix logic for collapsible options

guerler committed:
Fix type in form input wrapper related to connected modules

guerler committed:
Remove show call to connected/collapsible tooltip, not required for refresh

guerler committed:
Remove fixed width from parameter option icons

guerler committed:
Add connected check directly to associated value

mvdbeek committed:
Document and use mulled_channels option

mvdbeek committed:
Only parse build if build in name

Otherwise this fails with
```
Traceback (most recent call last):
  File "lib/galaxy/web/framework/decorators.py", line 277, in decorator
    rval = func(self, trans, *args, **kwargs)
  File "lib/galaxy/web/framework/decorators.py", line 101, in decorator
    return func(self, trans, *args, **kwargs)
  File "lib/galaxy/webapps/galaxy/api/tools.py", line 239, in install_dependencies
    tool._view.install_dependencies(tool.requirements, **kwds)
  File "lib/galaxy/tools/deps/views.py", line 107, in install_dependencies
    return self._dependency_manager._requirements_to_dependencies_dict(requirements, **kwds)
  File "lib/galaxy/tools/deps/__init__.py", line 190, in _requirements_to_dependencies_dict
    **kwds)
  File "lib/galaxy/tools/deps/container_resolvers/mulled.py", line 343, in resolve
    name = targets_to_mulled_name(targets=targets, hash_func=self.hash_func, namespace=self.namespace)
  File "lib/galaxy/tools/deps/container_resolvers/mulled.py", line 278, in targets_to_mulled_name
    version, build = split_tag(tag)
  File "lib/galaxy/tools/deps/mulled/util.py", line 69, in split_tag
    build = tag.split('--', 1)[1]
IndexError: list index out of range
```
if a single-target image was build with a build_mulled resolver.

jmchilton committed:
Fix dataset discovery for YAML/dictified tools.

mvdbeek committed:
Replace deprecated singularity import with build

And upgrade singularity build image to 2.4.6 (latest on biocontainers).
Overall this is necessary because singularity import 2.3X doesn't work
behind proxies. The change from `Singularity` to `Singularity.def` is
necessary because the script uses the `file` command if the Singularity
file doesn't end in .def or is not exactly Singularity (it is
`/import/Singularity` in our case). `file` isn't available in the
biocontainers image.

jmchilton committed:
Fix for parsing collection outputs for YAML/dictified tools.

Nerdinacan committed:
Added bundleEntries.getGalaxyInstance to tour generator, corrected similar global problem in library folder

Nerdinacan committed:
Merged dev, included store reference in bundleEntries

guerler committed:
Do not use default value in workflow execution form for runtime values

Nerdinacan committed:
Reimplemented global window.Galaxy catch for plugins

jmchilton committed:
Merge pull request #7155 from jmchilton/reuse_get_params

Small de-duplication cleanups to galaxy.jobs

nsoranzo committed:
Fix spelling in resolver docs

Co-Authored-By: mvdbeek <m.vandenbeek@gmail.com>

bgruening committed:
Merge pull request #7153 from jmchilton/fix_output_collection_yaml

Fix for parsing collection outputs for YAML/dictified tools.

bgruening committed:
Merge pull request #7152 from jmchilton/fix_yaml_discovery

Fix dataset discovery for YAML/dictified tools.

mvdbeek committed:
Allow anonymous downloading of test-data files

mvdbeek committed:
Also allow downloading test-data directories

This is (apparently ...) used by tools that create/test
directories in the extra_files_path.

mvdbeek committed:
Cache requests to mulled registry for 5 minutes

When going through the resolvers we can easily exceed the allowed
API request rate.

mvdbeek committed:
Move stream_archive to galaxy.util.streamball

mvdbeek committed:
Add `extra_files type="directory"` test

mvdbeek committed:
Make type="directory" remote compatible

We download an archive containing the specified directory
for tests.

mvdbeek committed:
Quote remaining parametera

mvdbeek committed:
More docker debug

jmchilton committed:
Merge pull request #7110 from jmchilton/os_to_dict

Add to_dict() to ObjectStores.

mvdbeek committed:
Test that test-data paths outside of test-data folder cannot be downloaded

mvdbeek committed:
Test singularity exec as Galaxy user

bernt-matthias committed:
doc: added missing docs for new UNIVA runner

mvdbeek committed:
Explicitly pass home dir to singularity command

mvdbeek committed:
Install singularity 3.0.1 from source

mvdbeek committed:
Bump image to use

jmchilton committed:
Merge pull request #7161 from bernt-matthias/topic/univa-doc

doc: added missing docs for new UNIVA runner

jmchilton committed:
Merge pull request #7160 from mvdbeek/unprivileged_remote_tool_tests

Unprivileged remote tool tests

jmchilton committed:
Merge pull request #7150 from guerler/connected_value

Follow-up on handling connectable and collapsible parameters in workflows

jmchilton committed:
Merge pull request #7124 from guerler/admin_error_log

Adds a vue component to display tool errors in the admin panel

jmchilton committed:
Merge pull request #7145 from guerler/fix_drop

Use regular js-event handling instead of jquery for uploader

mvdbeek committed:
Fix TestDataResolver.get_filecontent

machristie committed:
Lower provider name

The keys of the oidc_backends_config dict are created by lowercasing the
provider names in oidc_backends_config.xml file so we need to lower the
provider name given in the request in order to find the backend config.

jmchilton committed:
Merge pull request #7165 from mvdbeek/remote_test_data_followup

Fix TestDataResolver.get_filecontent

martenson committed:
Merge pull request #7146 from dannon/fix_toolshed_styles

Fixes toolshed and reports styling that was previously applied at the…

mvdbeek committed:
Use bash for mulled ContainerDescriptions

The base mulled container contains bash, and unfortunately
bashisms such as <(cat bla) are very common in widely used Galaxy
wrappers. So I think it makes sense to default to bash here.

mvdbeek committed:
Allow passing test_history in verify_tool func

nsoranzo committed:
Remove unnecessary split

nsoranzo committed:
Use dict ``setdefault()``

nsoranzo committed:
Remove mailto link from error reports

`email_str` is often not a simple email address.

nsoranzo committed:
Stricter ``process_key`` tests

mvdbeek committed:
Fix nesting and test

mvdbeek committed:
Fix test_test_data_download

This contains 5 items after
https://github.com/galaxyproject/galaxy-test-data/pull/2 had been
merged.

mvdbeek committed:
Merge pull request #7168 from nsoranzo/misc

Misc fixes

natefoo committed:
Unit tests for new DB serialization handler assignment methods

Nerdinacan committed:
Merged conflicting dev changes

nsoranzo committed:
Merge pull request #7171 from mvdbeek/fix_test_data_download_api_test

Fix test_test_data_download

nsoranzo committed:
Merge pull request #7169 from mvdbeek/fix_process_key_unordered_elements

Fix process_key test and deal with unordered repeat elements

natefoo committed:
Integration tests for handler assignment methods

bgruening committed:
Merge pull request #7166 from machristie/lower-provider-name

Lower provider name

jmchilton committed:
Merge pull request #7129 from natefoo/serializable-job-assignment

Perform job assignment using DB serialization techniques

mvdbeek committed:
Make dependency_shell_commands a property

If we know we use a container we don't need to pass
through the dependency resolver.

mvdbeek committed:
Introduce `auto_install` ContainerResolver flag

If set to True (thats's the default), the container
resolvers that can pull or build images will do that,
if they are enabled.
You can set this to `False` and instead pass through
the DependencyManager to use these resolvers to pull
or build images.

jmchilton committed:
Merge pull request #7157 from guerler/workflow_runtime_default_value

Do not use default values for runtime values in workflow execution form

nsoranzo committed:
Document ``message_box_class`` option

Reported by @vazovn

dannon committed:
Merge pull request #7174 from nsoranzo/message_box_class

Document ``message_box_class`` option

jmchilton committed:
Merge pull request #7125 from mvdbeek/MulledSingularityContainerResolver

Use singularity build / exec with biocontainers

jmchilton committed:
Update version to 19.01.rc1

jmchilton committed:
Update version to 19.05.dev

jmchilton committed:
Merge branch 'version-19.01' into version-19.05.dev

jmchilton committed:
Merge pull request #7144 from jmchilton/remove_collect_outputs_from

Remove collect_outputs_from option.

martenson committed:
prevent XSS on toolshed when browsing repo changeset

fix invalid changeset ts test, fix flake8

Revert "fix invalid changeset ts test, fix flake8"

This reverts commit 84864fe82a9f3b9ffa1cd5ef272f979e867e2002.

tweak the invalid changeset test and add a new one

dannon committed:
Fix hda accessibility check in Page's renderForSave name fetching

dannon committed:
Merge pull request #7128 from natefoo/slurm-trace-log-fix

[18.05] Fix for use of `log.trace` in Slurm runner

dannon committed:
Disable only-child padding manipulation on all #center divs

dannon committed:
Client format.

jmchilton committed:
Import regeneration test case.

dannon committed:
Avoid and log error on attempted redefinition of window.Galaxy

mvdbeek committed:
Add python to Count1 tool requirements

jmchilton committed:
Fix regeneration of metadata files for types besides bam during import.

dannon committed:
Drop unused registerGlobalInitFunctions file; this turned into globalInits/index

mvdbeek committed:
Transfer back testing image to galaxy namespace

nsoranzo committed:
Merge pull request #7176 from jmchilton/version-19.01

Update version to 19.01.rc1

nsoranzo committed:
Merge pull request #7177 from jmchilton/version-19.05.dev

Update to version 19.05.dev

jmchilton committed:
Merge pull request #7185 from mvdbeek/add_python_requirement

Add python to Count1 tool requirements

jmchilton committed:
Fix workflow parameter test case?

dannon committed:
Fix chart entry point libraries

nsoranzo committed:
Comment out unused line

simonbray committed:
adding itp and mdp datatypes (for gromacs) 

Resubmitting an old PR which somehow got lost: https://github.com/galaxyproject/galaxy/pull/6812/

simonbray committed:
adding itp and mdp datatypes (for gromacs) 

Resubmitting an old PR which somehow got lost: https://github.com/galaxyproject/galaxy/pull/6812/

jmchilton committed:
Merge pull request #7189 from simonbray/dev

Adding itp and mdp datatypes

dannon committed:
Incremental work getting trackster functional. -- panel management needs to be hooked up before we kick off the app

guerler committed:
Merge pull request #7188 from jmchilton/fix_wf_parameters

[19.01] Fix workflow parameter test case?

dannon committed:
Add galaxy_config back to global for this entrypoint; downstream uses detected...

dannon committed:
Fix viz views imports, don't assume global popupmenu

martenson committed:
fix library dataset permission API

broken in refactoring

martenson committed:
fix import ordering, thanks @nsoranzo

martenson committed:
refactor approach so the method is inherited to ldda

martenson committed:
call the method through the proper manager

Nerdinacan committed:
Added pages configuration object in editor.mako

dannon committed:
Move jquery.wymeditor attachment to jqueryCustom for consistency, and
other minor tweaks while digging into page error.

jmchilton committed:
Merge pull request #7190 from martenson/fix-permission-api

[18.09] fix library dataset permission API

dannon committed:
Merge pull request #7180 from martenson/security-ts

[18.09] Prevent XSS on toolshed when browsing repo changeset

jmchilton committed:
Fix error handling in export_history jobs.

Broken since initial commit (e1876ef7670ac1ed1ee21255572a556de1fe954a).

mvdbeek committed:
Make interactor changes backwards compatible

nsoranzo committed:
Merge pull request #7192 from jmchilton/fix_export_error_handling

Fix error handling in export_history jobs.

dannon committed:
Merge pull request #7117 from Nerdinacan/init_queue_2

Client initialization pipeline

mvdbeek committed:
Merge pull request #7186 from jmchilton/import_metadata_files

Fix regeneration of metadata files for types besides bam during import.

martenson committed:
Merge pull request #7167 from mvdbeek/verify_tool_improvements

Remote tool testing enhancements

nsoranzo committed:
Merge branch 'release_18.05' into release_18.09

nsoranzo committed:
Merge branch 'release_18.09' into release_19.01

nsoranzo committed:
Merge branch 'release_19.01' into dev

dannon committed:
Re-enable gulp task transpiling/staging some galaxy code, pending a more
thorough review of external accessors.  This fixes (at minimum) IE
launching.

dannon committed:
Uglify staged scripts used by external accessors.

dannon committed:
Leave TODO for removing these particular tasks prior to 19.05

dannon committed:
Re-stage largespinner, since there are external accessors...

dannon committed:
Since base.css (packed) is available, use those styles instead of staging manually.

jxtx committed:
Merge pull request #7196 from dannon/restage_scripts

[19.01] Re-enable script staging

dannon committed:
Merge remote-tracking branch 'upstream/release_19.01' into dev

blankenberg committed:
Merge pull request #7187 from mvdbeek/transfer_back_docker_image

Transfer back testing image to galaxy namespace

martenson committed:
Merge pull request #7181 from dannon/page_save_history_fix

[18.05] Fix hda accessibility check in Page's renderForSave name fetching

martenson committed:
Merge branch 'release_18.05' into release_18.09

martenson committed:
Merge branch 'release_18.09' into dev

martenson committed:
Merge branch 'tool_favs' of github.com:martenson/galaxy into tool_favs

martenson commented:
superseded by https://github.com/galaxyproject/galaxy/pull/7209
